libbpf-tools/runqlen.c

```diff
@@ -222,22 +235,22 @@ int main(int argc, char **argv)
 		return 1;
 	}
 
-	obj = runqlen_bpf__open();
-	if (!obj) {
-		fprintf(stderr, "failed to open and/or load BPF object\n");
+	nr_cpus = libbpf_num_possible_cpus();
+	if (nr_cpus < 0) {
+		printf("failed to get # of possible cpus: '%s'!\n",
+		       strerror(-nr_cpus));
 		return 1;
 	}
-
-	nr_cpus = libbpf_num_possible_cpus();
 	if (nr_cpus > MAX_CPU_NR) {
-		fprintf(stderr, "The number of cpu cores is too much, please "
+		fprintf(stderr, "the number of cpu cores is too much, please "
 			"increase MAX_CPU_NR's value and recompile");
 		return 1;
 	}
-	links = calloc(nr_cpus, sizeof(*links));
-	if (!links) {
-		fprintf(stderr, "failed to alloc links\n");
-		goto cleanup;
+
+	obj = runqlen_bpf__open();
+	if (!obj) {
+		fprintf(stderr, "failed to open and/or load BPF object\n");
```

Comment:
just open, not yet load

Reply:
Fixed, thanks! I'll check other tools as well and fix them in next PR.